imod/mf6/wel.py

```diff
@@ -220,31 +231,50 @@ def clip_box(
         ----------
         time_min: optional
         time_max: optional
-        z_min: optional, float
-        z_max: optional, float
+        layer_min: optional, int
+        layer_max: optional, int
         x_min: optional, float
         x_max: optional, float
         y_min: optional, float
         y_max: optional, float
+        top: optional, GridDataArray
+        bottom: optional, GridDataArray
+        state_for_boundary: optional, GridDataArray
 
         Returns
         -------
         sliced : Package
         """
+        if (layer_max or layer_min) and (top is None or bottom is None):
+            raise ValueError("Both the layer_max and the top should be defined")
+
+        if top is not None:
+            if not isinstance(top, GridDataArray) or "layer" not in top.coords:
+                top = create_layered_top(bottom, top)
 
         # The super method will select in the time dimension without issues.
         new = super().clip_box(time_min=time_min, time_max=time_max)
 
         ds = new.dataset
 
+        z_max = None if layer_max is None else top.isel(layer=layer_max)
+        z_min = None if layer_min is None else bottom.isel(layer=layer_min)
+
+        if z_max:
+            ds["screen_top"] = ds["screen_top"].clip(None, z_max)
+        if z_min:
+            ds["screen_bottom"] = ds["screen_bottom"].clip(z_min, None)
```

Comment:
I don't think this call to ``clip`` works in the case that ``top`` and ``bottom`` is a grid, since wells are not defined as a grid.